caluma/caluma_core/health_checks.py

```diff
@@ -0,0 +1,213 @@
+from django.conf import settings
+from django.db import connections
+from django.db.migrations.executor import MigrationExecutor
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.apps.registry import apps
+from django.http import JsonResponse
+from django.core import management
+
+from caluma.caluma_logging.models import AccessLog
+from caluma.caluma_form import storage_clients
+
+import watchman
+from watchman import views
+from watchman.decorators import check
+from watchman import settings as watchman_settings
+
+from io import BytesIO, StringIO
+import json
+
+
+@check
+def _check_pending_migrations(db):
+    """
+    Check database for pending migrations.
+    Returns JSON mapping if no pending migrations, otherwise raises Exception.
+    @check decorator catches and handles exception.
+    """
+
+    # check for changes i.e. migrations that haven't been made
+    # makemigrations --check exists with exit status 1 if changes exist
+    try:
+        management.call_command("makemigrations", "--check", "--dry-run")
+    except SystemExit:
+        raise Exception("Migrations aren't up-to-date (makemigrations).")
+
+    # check for unapplied migrations
+    try:
+        out = StringIO()
+        management.call_command("migrate", "--plan", stdout=out)
+        plan = out.getvalue()
+        print(plan)
+        if not "No planned migration operations." in plan:
+            raise Exception("Database has unapplied migrations (migrate).")
+
+        return {db: {"ok": True}}
+
+    finally:
+        # Cleanup
+        out.close()
```

Comment:
Good thinking, but not really needed - The garbage collector will do the work for you anyway.

Reply:
Does one never close files etc. in Python because the garbage collector eventually takes care of it? Or it's just not necessary to double check in a `finally` block?

Reply:
The "proper" way would be to use a `with` block, but in this context it's not really usable. I'd do an explicit `close()` if we need to assume that other parts of the code may have a reference to the file, but we do want it to be closed. But that's not the case here, so we can trust the GC to do it for us